code/lighting/lighting_profiles.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "globalincs/vmallocator.h"
+
+enum TonemapperAlgorithm :int {
+	tnm_Invalid = -1,
+	tnm_Linear = 0,
+	tnm_Uncharted = 1,
+	tnm_Aces = 2,
+	tnm_Aces_Approx = 3,
+	tnm_Cineon = 4,
+	tnm_Reinhard_Jodie = 5,
+	tnm_Reinhard_Extended = 6,
+	tnm_PPC = 7,
+	tnm_PPC_RGB= 8
+};
+
+struct piecewise_power_curve_values{
+	float toe_strength;
+	float toe_length;
+	float shoulder_strength;
+	float shoulder_length;
+	float shoulder_angle;
+};
+
+class lighting_profile{
+public:
+	static enum TonemapperAlgorithm name_to_tonemapper(SCP_string &name);
+	static void load_profiles();
+	static TonemapperAlgorithm current_tonemapper();
+	static piecewise_power_curve_values current_piecewise_values();
```

Comment:
I'd suggest for this to return a ``const piecewise_power_curve_values &``.

Reply:
Correct me if I'm wrong, but that would the calling code access to the actual underlying data structure, while the current code creates and returns a copy of the struct's values? I would prefer to return only the values and make sure nothing can accidentally tweak the profile data in the future.

Reply:
What this does is return a reference to the values of the underlying data structure, preventing an unnecessary copy.
But since you return a ``const`` reference and not just a reference, it cannot be used to modify the underlying data, only allowing read-only access. Since this both prevents modification and unnecessary copying, returning by const & should always be preferable to returning by copy unless you either need the copy every time, or you return something that's allocated on the stack (a variable declared within the function). Since neither is true here, you can and should return a const reference.

Reply:
Okay, that makes sense then.